### PR TITLE
merge state and value changes for input changes

### PR DESCRIPTION
**Pull request by bcandullo**
**Jira**
https://jira.inbcu.com/browse/AUG-957

**Changes**
- <del>expose public method to mark change detector on clearable icon component</del>
- have `ClearableIconComponent` subscribe to state changes as well as value changes to detect changes

**Peers**
https://github.com/NBCUAS/audri/pull/395

### CODE REVIEW

File: src/lib/clearable-input/clearable-input.directive.ts [hunk 1]
```diff
@@ -113,4 +113,13 @@ export class ClearableInputDirective implements OnInit, OnDestroy {
         }
       });
   }
+
+  /*
+  * Explicitly force icon component to get checked.
+  */
+  public updateClearableIconChanges() {
```
Comment (bakesaled) [inline src/lib/clearable-input/clearable-input.directive.ts]: While this seems to solve the problem, I'm wondering if there is a deeper issue here.  Given that the ClearableIcon component listens for the valueChange event to know when to swap icons, I'm thinking the underlying input component is not emitting valueChange when the value property is set.

Comment (bcandullo) [reply src/lib/clearable-input/clearable-input.directive.ts]: This is correct. When I emit a value change here:
https://github.com/CINBCUniversal/suprematism/blob/master/src/lib/input/input.component.ts#L56

Everything is good. Should we make the icon input pick up state changes as well? I'm talking about this: https://github.com/CINBCUniversal/suprematism/blob/master/src/lib/clearable-input/clearable-icon/clearable-icon.component.ts#L77

Comment (bakesaled) [reply src/lib/clearable-input/clearable-input.directive.ts]: Yeah, that might work.

Comment (bcandullo) [reply src/lib/clearable-input/clearable-input.directive.ts]: Yep, it works with all of the use cases I have 👍

**Pull request merged**